test/tools/libtesteth/boostTest.cpp

```diff
@@ -137,6 +149,38 @@ int main(int argc, const char* argv[])
     int result = 0;
     auto fakeInit = [](int, char* []) -> boost::unit_test::test_suite* { return nullptr; };
     result = unit_test_main(fakeInit, argc, const_cast<char**>(argv));
+
+    // Print suggestions of a test case
+    if (result == 200)  // test suite not found
+    {
+        typedef std::pair<string, size_t> NameDistance;
+        std::vector<NameDistance> distanceMap;  // <index in availableTests, compared distance>
+        for (auto& it : allTests)
+        {
+            int dist =
+                test::levenshtein_distance(sMinusTArg.c_str(), sMinusTArg.size(), it, strlen(it));
+            distanceMap.push_back({it, dist});
+        }
+        std::sort(distanceMap.begin(), distanceMap.end(),
+            [](NameDistance const& a, NameDistance const& b) { return a.second < b.second; });
+        std::cerr << "Did you mean: "
+                  << "\n";
+        std::set<string> suggestionSet;
+        size_t hasSlash = sMinusTArg.find_first_of("/");
+        for (size_t i = 0; i < 3; i++)
+        {
+            if (i < distanceMap.size())
+            {
+                string name = distanceMap[i].first;
+                if (hasSlash ==
```

Comment:
Ah no, it could be several test suites with similar names I guess, and this wouldn't cover it...

Maybe it's fine to show only one top suggestion for the test suite?
Using `set` really isn't suitable here, it seems to me.